projects/packages/videopress/src/client/block-editor/blocks/video/components/player/index.native.js

```diff
@@ -63,20 +73,72 @@ export default function Player( { isSelected, attributes } ) {
 		poster,
 	} );
 
-	const { preview, isRequestingEmbedPreview } = usePreview( videoPressUrl );
-	const html = addTokenIntoIframeSource( preview?.html, token );
+	// Helper to invalidate the preview cache.
+	const invalidateResolution = useDispatch( coreStore ).invalidateResolution;
+	const invalidatePreview = useCallback( () => {
+		invalidateResolution( 'getEmbedPreview', [ videoPressUrl ] );
+	}, [ videoPressUrl, invalidateResolution ] );
+
+	const { preview = {}, isRequestingEmbedPreview } = usePreview( videoPressUrl );
+	const html = addTokenIntoIframeSource( preview.html, token );
+
+	const cancelPreviewCheckInterval = useCallback( () => {
+		clearInterval( previewCheckInterval.current );
+	}, [ previewCheckInterval ] );
+
+	// Fetch the preview until it's ready
+	useEffect( () => {
+		// Wait for the client player to load
+		if ( ! isPlayerLoaded ) {
+			return;
+		}
+		previewCheckInterval.current = setInterval( () => {
+			// if the preview is ready or we reached the max attempts, clear the interval
+			if ( isPreviewReady( preview ) || previewCheckAttempts > VIDEO_PREVIEW_ATTEMPTS_LIMIT ) {
+				return cancelPreviewCheckInterval();
+			}
+
+			// Avoid lapping any prior request
+			if ( isRequestingEmbedPreview ) {
```

Comment:
I considered having the counter increment before this early return. I left it below since an ongoing request has not been evaluated and should not count as an attempt.